unit-tests/func/alt-ir/alt-ir-common.h

```diff
@@ -0,0 +1,111 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include "../../test.h"
+#include <concurrency.h>
+
+using namespace rs2;
+
+void set_alt_ir_if_needed( const rs2::depth_sensor & depth_sens, float val )
+{
+    // set alt_ir if it was disabled
+    REQUIRE( depth_sens.supports( RS2_OPTION_ALTERNATE_IR ) );
+    auto alt_ir = depth_sens.get_option( RS2_OPTION_ALTERNATE_IR );
+    if( alt_ir != val )
+        REQUIRE_NOTHROW( depth_sens.set_option( RS2_OPTION_ALTERNATE_IR, val ) );
+}
+
+void enable_alt_ir_and_check_that_AC_fails(
+    const rs2::device & dev,
+    const rs2::depth_sensor & depth_sens,
+    const std::vector< stream_profile > & expected_profiles )
+{
+    std::vector< stream_profile > profiles = expected_profiles;
+    REQUIRE_NOTHROW( depth_sens.open( profiles ) );
+
+    REQUIRE_NOTHROW( depth_sens.start( [&]( rs2::frame f ) {} ) );
+
+    set_alt_ir_if_needed( depth_sens, 1 );
+
+    // check that AC throws exception
+    if( dev.is< rs2::device_calibration >() )
+    {
+        std::condition_variable cv;
```

Comment:
Technically, the exception (and therefore status) gets throw even before we exit `trigger_device_calibration` -- meaning we have a status and all this locking mechanism is not really needed?

Reply:
done